lib/formtastic/helpers/inputs_helper.rb

```diff
@@ -350,12 +350,21 @@ def association_columns(*by_associations) # @private
         end
       end
 
+      # Collects non-standard foreign key columns
+      def foreign_key_columns
+        if @object.present? && @object.class.respond_to?(:reflections)
+          @object.class.reflections.map {|_,v| v.options[:foreign_key] }.compact
```

Comment:
I see what you mean. Formtastic actually does not depend on ActiveRecord. But then it is kind of weak dependency as those methods like `content_columns` are from ActiveRecord.

I'd say it is fine to add ActiveRecord as a test dependency, because we actually want to test against ActiveRecord. I'll start a branch soon exploring that and replacing few mocks with ActiveRecord objects.